src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateStringUpdater.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.TemplateEngine.TemplateLocalizer.Core
+{
+    internal static class TemplateStringUpdater
+    {
+        /// <summary>
+        /// Updates the templatestrings.json files for given languages with the provided strings.
+        /// </summary>
+        /// <param name="strings">Template strings to be included in the templatestrings.json files.</param>
+        /// <param name="templateJsonLanguage">The language of the <paramref name="strings"/>.</param>
+        /// <param name="supportedLanguages">The list of languages for which templatestrings.json file will be created.</param>
+        /// <param name="targetDirectory">The directory that will contain the generated templatestrings.json files.</param>
+        /// <param name="logger"><see cref="ILogger"/> to be used for logging.</param>
+        /// <param name="cancellationToken">Cancellation token to cancel the operation.</param>
+        /// <returns>The task that tracks the status of the async operation.</returns>
+        public static async Task UpdateStringsAsync(
+            IEnumerable<TemplateString> strings,
+            string templateJsonLanguage,
+            IEnumerable<string> supportedLanguages,
+            string targetDirectory,
+            ILogger logger,
+            CancellationToken cancellationToken)
+        {
+            Directory.CreateDirectory(targetDirectory);
+
+            foreach (string language in supportedLanguages)
+            {
+                string locFilePath = Path.Combine(targetDirectory, language + ".templatestrings.json");
+                Dictionary<string, string>? existingStrings = null;
+
+                // Check existing strings only if this is not the original language the template was created in.
+                // Because we know that the text in template.json is always more up-to-date than the text in templatestrings.json.
+                if (templateJsonLanguage != language)
```

Comment:
should we create the localization file for native language of template.json? 
I believe it will only impact performance, as reading from template.json is faster than from locale file. If certain locale is missing, anyway template.json will be used.

Reply:
I will keep this as is after the internal discussions we had. The file will be ignored at runtime by the engine instead.